x-pack/plugin/ml/qa/ml-with-security/build.gradle

```diff
@@ -239,6 +239,7 @@ tasks.named("yamlRestTest").configure {
     'ml/set_upgrade_mode/Setting upgrade_mode to enabled',
     'ml/set_upgrade_mode/Setting upgrade mode to disabled from enabled',
     'ml/set_upgrade_mode/Test setting upgrade_mode to false when it is already false',
+    'ml/text_embedding_search/Test text embedding search',
```

Comment:
I think if you call the index used for testing `index-with-embedded-text ` instead of `embedded_text` then it won't be necessary to skip the test here, and that will also be a way to get the test framework to confirm that users who don't have ML permissions can use semantic search if they have access to the index being searched.

This code will also need to be deleted: https://github.com/elastic/elasticsearch/blob/6ccaa94c9fedf257922e71fc6040aa97814ca7c4/x-pack/plugin/ml/qa/ml-with-security/src/yamlRestTest/java/org/elasticsearch/smoketest/MlWithSecurityInsufficientRoleIT.java#L47-L58

That assertion must predate the decision that semantic search wouldn't require permission to call inference directly, and is inconsistent with it.